drivers/dma/dma_mcux_lpc.c

```diff
@@ -193,39 +200,110 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 				config->block_count);
 			return -EINVAL;
 		}
-		int i = 0;
-		dma_transfer_config_t *next_transfer;
+
+		/* Allocate the descriptor table structures if needed */
+		if (data->dma_descriptor_table == NULL) {
+			data->dma_descriptor_table = k_malloc(CONFIG_DMA_LINK_QUEUE_SIZE *
+							(sizeof(dma_descriptor_t) +
+							 FSL_FEATURE_DMA_LINK_DESCRIPTOR_ALIGN_SIZE));
+
+			if (!data->dma_descriptor_table) {
+				LOG_ERR("HEAP_MEM_POOL_SIZE is too small");
+				return -ENOMEM;
+			}
+			/* Ensure descriptor is aligned */
+			data->dma_descriptor_table = (dma_descriptor_t *)ROUND_UP(
+								data->dma_descriptor_table,
+								FSL_FEATURE_DMA_LINK_DESCRIPTOR_ALIGN_SIZE);
+		}
+
+		/* Prepare the transfer using the first block, queue more
+		 * descriptors as needed
+		 */
+		DMA_PrepareTransfer(
+			&xferConfig,
+			(void *)block_config->source_address,
+			(void *)block_config->dest_address,
+			config->dest_data_size,
+			block_config->block_size, transfer_type,
+			(void *)data->dma_descriptor_table);
+
+		/* Disable interrupts for first descriptors to avoid calling
+		 * callback multiple times.
+		 */
+		xferConfig.xfercfg.intA = false;
+		xferConfig.xfercfg.intB = false;
+
+		dma_descriptor_t *next_transfer;
+		dma_descriptor_t *curr_transfer = data->dma_descriptor_table;
+		dma_xfercfg_t tmp_xfercfg;
+		uint32_t dest_width = config->dest_data_size;
+
+		block_config = block_config->next_block;
 
 		while (block_config != NULL) {
 			/* Check if this last element in the chain */
 			if (block_config->next_block == NULL) {
 				next_transfer = NULL;
 			} else {
-				next_transfer = &(data->transferConfig[i + 1]);
+				next_transfer = curr_transfer + sizeof(dma_descriptor_t);
+				/* Ensure descriptor is aligned */
+				next_transfer = (dma_descriptor_t *)ROUND_UP(
```

Comment:
this could be a problem if size of the dma_descriptor_t is not 16 bytes aligned, as we malloc only  one 
FSL_FEATURE_DMA_LINK_DESCRIPTOR_ALIGN_SIZE additional size, if we patched alignment each loop then this seems not enough.

Reply:
The amount of memory allocated in k_malloc accounts for this. Below is memory size allocated `(CONFIG_DMA_LINK_QUEUE_SIZE * (sizeof(dma_descriptor_t) + FSL_FEATURE_DMA_LINK_DESCRIPTOR_ALIGN_SIZE))`

Reply:
ok, right. I miss the `)`